x/ibc/03-connection/keeper/handshake.go

```diff
@@ -165,32 +166,17 @@ func (k Keeper) ConnOpenAck(
 	}
 
 	// Check connection on ChainA is on correct state: INIT or TRYOPEN
-	if connection.State != types.INIT && connection.State != types.TRYOPEN {
+	// If the state is in INIT the encoded version must be supported.
+	// If the state is in TRYOPEN the encoded version must be the only
+	// set version in the retreived connection state.
+	if !((connection.State == types.INIT && types.IsSupportedVersion(encodedVersion)) ||
+		(connection.State == types.TRYOPEN && len(connection.Versions) == 1 && connection.Versions[0] == encodedVersion)) {
```

Comment:
nit, this is hard to read. Can we assign a variable name for each case (init and tryopen)?

Reply:
I could alternatively do a swtich

```
switch {
case INIT && !supported:
    return err
case TRYOPEN && wrongVersion:
    return err
}
```

Reply:
@fedekunze I update this to use a switch. I think it is more clear, let me know if any adjustments should be made